token/js/client/token.js

```diff
@@ -2239,6 +2239,9 @@ export class Token {
     mint: PublicKey,
     owner: PublicKey,
   ): Promise<PublicKey> {
+    if (!PublicKey.isOnCurve(owner.toBuffer())) {
```

Comment:
some thoughts on the reason of this: https://github.com/solana-labs/solana-program-library/pull/1802#discussion_r640742549